src/EnergyPlus/InputProcessing/InputProcessor.cc

```diff
@@ -466,6 +466,18 @@ bool InputProcessor::checkForUnsupportedObjects(EnergyPlusData &state)
     return errorsFound;
 }
 
+void InputProcessor::getSimInputObjects(EnergyPlusData &state)
+{
+    // there is a need to know if certain object are used in the simulation
+    // this may not be the best place to do this but it's a start at early reading of input data
+    // this concept could grow to include reading all inputs prior to the start of the simulation so all data is available
+    // once inputs are processed, read in all getInputs. They will be read early and will not be read again so there is no duplication
+
+    // for example, AirLoopHVAC:DOAS system autosize on weather data, and WeatherManager processes that data
+    // there is no need to process that data if there are no DOAS used in the simulation
+    state.dataInputProcessing->DOASUsedInSim = (state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, "AirLoopHVAC:DedicatedOutdoorAirSystem") > 0);
+}
+
```

Comment:
To avoid unnecessary code execution in weather manager I added a bool. Chose input processor for the location since that's what this call does. Might want a different location since my expectation is to read in all getInputs prior to sizing. It doesn't matter from an execution time whether getInputs are read early or when the model is called. But from a code order perspective the data needed to process certain objects has to wait on other objects to be read in. An example is UnitarySystem in an air loop where information on zone equipment is needed. If that data were already available the UnitarySystem could process all getInput items in a one-shot. There are other reasons to do this, e.g., if (ZoneEquipmentReadIn) or like statements scattered throughout code. So this is the thought behind this new function,  living here are other location. If this will stay here I will clang format, otherwise that simple change would be a drain on the CI. So what-say-you on this location? and suggestion for other location if needed.

Reply:
I guess input processor makes sense. I'm not sure about calling this `getSimInputObjects`. That seems a bit premature, especially since it won't be getting an object, just checking how many are present. 

But maybe all of these could be called from a single function along with this new check?
```
        CheckIfAnyEMS(state);
        CheckIfAnyPlant(state);
        CheckIfAnySlabs(state);
        CheckIfAnyBasements(state);
        CheckIfAnyIdealCondEntSetPoint(state);
```

Reply:
Now I think I should have put this new function in Data\EnergyPlusData. Already has all #include needed, is a global access point, and out of the way, and manager, of all other classes. Not for today, but I'm just not sure inputProcessor is the right place.

Reply:
That is the same thought I had about InputProcessor. A new module? existing?